CLIENT-SPECIFICATION.md

```diff
@@ -38,7 +38,8 @@ Option             | Required?   | Meaning
 `-v`, `--version`  | Yes         | Shows the current version of the client, and the version of this specification that it implements.
 `-p`, `--platform` | Yes         | Specifies the platform to be used to perform the action (either listing or searching) as an argument. If this option is specified, the selected platform MUST be checked first instead of the current platform as described below.
 `-u`, `--update`   | Conditional | Updates the offline cache of pages. MUST be implemented if cache is supported.
-`-l`, `--list`     | No          | Lists all the pages in the current platform to the standard output. If the special platform `all` is specified a list of all pages in all platforms MUST be displayed.
+`-l`, `--list`     | No          | Lists all the pages in the current platform to the standard output. It is RECOMMENDED that the pages are sorted alphanumerically.
+      `--list-all` | No          | Lists all the pages in all platforms to the standard output. It is RECOMMENDED that the pages are sorted alphanumerically.
```

Comment:
Simply specifying sorting alphabetical ordering isn't always though. [As Jeff Attwood explains](https://blog.codinghorror.com/sorting-for-humans-natural-sort-order/), unicode can cause an issue here. We should find some standard on unicode sorting here.

The other thing we should handle here is the case of when a page exists in multiple platforms. Perhaps we should require the platform name to be specified as well here, and explicitly state that a given page name may exist in multiple categories and that's ok and they `MUST NOT` be deduplicated?

Reply:
Yeah, you're right. As written in my other comment above, after some thought I'd actually tend towards not adding `--list-all` to the standard at all. Same thing with `--list`, we could leave it to the implementation to choose a useful output format. (Or is there any advantage in having a common output format? Should it be parseable?)

Reply:
Yeah, I think that's a good point. Clients can be for many different platforms and purposes - even if we assume a command line, there's a range of possible purposes that such a client may be implemented for. Let's drop the sorting sentence here.

Reply:
I pushed a new version that simply removed the `all` platform and doesn't specify anything else. Implementations can still implement `--list-all` if they want, but if the output format isn't standardized, then a standardized command name won't bring any benefits either.

Reply:
Right. Let's add support for `--list-all`. What about specifying 1 command name per line on the standard output? e.g.

```markdown
`--list-all` | No          | Lists all the pages in all platforms to the standard output, with 1 page name per line in the format `<platform>/<pagename>`
```

Reply:
@sbrl did you take a look at [this comment](https://github.com/tldr-pages/tldr/pull/7561#issuecomment-997473651)? I think adding `--list-all` to the spec isn't a good idea, there are a lot of open questions and a standardized `--list-all` wouldn't really give a lot of benefits while potentially limiting improvements to the output format.

My curent suggestion would be dropping the `all` platform and _not_ standardizing `--list-all`. I renamed the PR to reflect the current state of the proposal. If you agree, this discussion can be marked as resolved.